snuba/consumer.py

```diff
@@ -4,45 +4,71 @@
 
 from batching_kafka_consumer import AbstractBatchWorker
 
-from . import processor
-from .writer import row_from_processed_event, write_rows
+from .writer import write_rows
 
 
 logger = logging.getLogger('snuba.consumer')
 
 
+class InvalidMessageType(Exception):
+    pass
+
+
+class InvalidMessageVersion(Exception):
+    pass
+
+
 class InvalidActionType(Exception):
     pass
 
 
+# action types
+INSERT = object()
+REPLACE = object()
+
 class ConsumerWorker(AbstractBatchWorker):
-    def __init__(self, clickhouse, dist_table_name, producer, replacements_topic, metrics=None):
+    """
+    ConsumerWorker processes the raw stream of events coming
+    from Kafka, validates that they have the correct format
+    and does one of 2 things:
+
+        - If they are INSERTs, creates rows to insert from them
+          and batches and inserts those rows to clickhouse.
+        - If they are REPLACEs, forwards them to the replacements
+          Kafka topic, so that the Replacements consumer can deal
+          with them.
+    """
+    def __init__(self, clickhouse, dataset, producer, replacements_topic, metrics=None):
         self.clickhouse = clickhouse
-        self.dist_table_name = dist_table_name
+        self.dataset = dataset
         self.producer = producer
         self.replacements_topic = replacements_topic
         self.metrics = metrics
 
     def process_message(self, message):
         value = json.loads(message.value())
 
-        processed = processor.process_message(value)
-        if processed is None:
+        validated = self.validate_message(value)
```

Comment:
@bretthoerner I think this is one of the main parts where I actually changed stuff as opposed to just moving it around. Basically trying to make this `ConsumerWorker` be as generic as possible, and only validate the format of the outer payload `(version, command, data)`. Then pass it so a dataset-specific processor to be processed into a row.